pkg/sql/alter_table.go

```diff
@@ -306,7 +306,12 @@ func (n *alterTableNode) Start(params runParams) error {
 				}
 			}
 			if !found {
-				return fmt.Errorf("column %q in the middle of being added, try again later", t.Column)
+				for _, mutation := range n.tableDesc.Mutations {
+					if mutation.GetColumn().ID == col.ID {
```

Comment:
the column needs to be a mutation and the mutationID must be part of p.session.TxnState.schemaChangers.schemaChangers (see my comment below)

Once that's true you can safely remove it from the mutation list through

n.tableDesc.Mutations = append(n.tableDesc.Mutations[:i], n.tableDesc.Mutations[i+1:]...)